src/DatePicker/DatePicker.web.js

```diff
@@ -1,52 +1,188 @@
 // @flow
 
 import * as React from 'react';
-import DatePicker from 'react-datepicker';
-import 'react-datepicker/dist/react-datepicker.css';
+import { View, TouchableWithoutFeedback } from 'react-native';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import { StyleSheet } from '../PlatformStyleSheet';
+import DatePickerDayTile from './DatePickerDayTile';
+import { Button } from '../Button';
+import { Text } from '../Text';
 
 import type { Props } from './DatePickerTypes';
 
 type State = {
   date: Date,
+  month: number,
+  year: number,
+};
+
+const parsePropsToState = date => {
+  const tempDate = date ?? new Date();
+  return {
+    date: tempDate,
+    month: tempDate.getMonth(),
+    year: tempDate.getFullYear(),
+  };
 };
 
 export default class WebDatePicker extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
 
-    this.state = {
-      date: props.date || new Date(),
-    };
+    this.state = parsePropsToState(props.date);
   }
 
-  handleConfirm = (date: Date) => {
+  componentDidUpdate = (prevProps: Props) => {
+    const { date } = this.props;
+    if (date && date !== prevProps.date) {
+      this.setState(parsePropsToState(date));
+    }
+  };
+
+  handleDaySelect = (dayId: number) => {
     const { onConfirm } = this.props;
+    const { month, year } = this.state;
+    const date = new Date(year, month, dayId);
+    this.setState(parsePropsToState(date));
+    onConfirm(date);
+  };
 
-    this.setState({
-      date,
+  swapMonthBack = () => {
+    this.setState(state => {
+      const month = state.month === 0 ? 11 : state.month - 1;
```

Comment:
Could these helper date functions be moved to a different file and can they be tested too?

Reply:
Moved and tests added.